internal/service/route53/traffic_policy.go

```diff
@@ -136,23 +136,20 @@ func resourceTrafficPolicyRead(ctx context.Context, d *schema.ResourceData, meta
 	return diags
 }
 
-func resourceTrafficPolicyUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+func resourceTrafficPolicyVersionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	var diags diag.Diagnostics
 	conn := meta.(*conns.AWSClient).Route53Client(ctx)
 
-	input := &route53.UpdateTrafficPolicyCommentInput{
-		Id:      aws.String(d.Id()),
-		Version: aws.Int32(int32(d.Get(names.AttrVersion).(int))),
-	}
-
-	if d.HasChange(names.AttrComment) {
-		input.Comment = aws.String(d.Get(names.AttrComment).(string))
+	input := &route53.CreateTrafficPolicyVersionInput{
+		Document: aws.String(d.Get("document").(string)),
+		Id:       aws.String(d.Id()),
+		Comment:  aws.String(d.Get(names.AttrComment).(string)),
 	}
 
-	_, err := conn.UpdateTrafficPolicyComment(ctx, input)
+	_, err := conn.CreateTrafficPolicyVersion(ctx, input)
```

Comment:
Unfortunately this is not the full solution.
It should read the output - esp. new version ID and save it on the resource data so it's marked as modified. 

In other words we need to inform `aws_route53_traffic_policy_instance` the a new version got created.



I tried sth like but it doesn't fully work:
```diff
-       _, err := conn.CreateTrafficPolicyVersion(ctx, input)
+       output, err := conn.CreateTrafficPolicyVersion(ctx, input)
+       d.Set(names.AttrVersion, output.TrafficPolicy.Version)
```

Reply:
A similar case for aws secret version -> https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/secretsmanager/secret_version.go#L116 
It concats ID with version ID. Writing the code that changes the resource identifier that is backward-compatible might be hard.